intrastat_product/models/account_move.py

```diff
@@ -67,9 +67,16 @@ class AccountMoveLine(models.Model):
     hs_code_id = fields.Many2one(
         comodel_name="hs.code", string="Intrastat Code", ondelete="restrict"
     )
+    weight = fields.Integer(string="Weight", help="Net weight in Kg")
```

Comment:
Why do you need these extra fields. I think it's too much, and more not being even computed.

Reply:
@pedrobaeza 
They should indeed not being computed since this is to encode an invoice without product records (hence only intrastat code, weight & country of origin) or in case of product records with unknown weight (product with variable weight, hence weighting operation needed).

Reply:
But I don't get why this fixes the stack trace.

Reply:
@pedrobaeza 

The stack trace is fixed by the "if not product::" code block (cf. PR line 275 of the declaration)

Reply:
Then please limit this PR to that code, an propose improvements into another for debating such improvement.

Reply:
Not really in this case since the fix would be completely different (disallow entering intrastat code without product record). This feature should have been there for a long time since (at least on our market) common accounting packages support this. With the optional="hide" in Odoo 13.0 we can do this in a rather clean way.

Reply:
But it doesn't make sense to put something without product. For me it should be required if you want Intrastat.

Reply:
Oh yes it does.

Assume you are importing on a regular basis goods from EU countries which makes your company subject to the intrastat arrival declaration.
For the goods that you are importing on a regular basis you will work with product records, purchase orders, ...
But for some other goods you may work without purchase orders or a generic product record.
If you receive your supplier invoice it is a common practice that the intrastat code is on the invoice lines (if not than you need to take the shipping docs).
You don't want to do the overhead of creating a product record with weight for a one time purchase hence you simply encode the intrastat code and weight on the invoice line.
Especially for companies migrating from an accounting package having this feature, this is a must have (this particular PR is driven by a customer who migrated from winbooks, which is the leading accounting package on the belgian market).

The other scenario is for products with variable weight. The product code is always the same but the weight is different per transaction. Hence you need to take the shipping docs when encoding the invoice and enter the weight in order to generate your intrastat declaration.

Reply:
And why this hasn't been a problem till v13? As said, this is an improvement, and not a bug fix, and it should be debated in a separate PR. We have to avoid the stack trace, and the bugfix is to make the product required. 

All the rest for me should be debated in a separate PR, but you already know that I don't want to bring such low level data to each account move where inconsistency is going to happen for sure:

- Incorrect data in the product, that is propagated to lines
- On previous versions, you fix it on the product, and everything is OK. Not with this change.